### PR TITLE
Raise GitterRoomError by default if room not found

### PR DESCRIPTION
**Pull request by danmichaelo**
When trying to send a message to a room, this happened:

```
gitter.messages.send('org-room/Lobby', 'Bleep bloop')
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
<ipython-input-4-5ac87798ac23> in <module>()
----> 1 gitter.messages.send('wikimedia-croptool/lobby', 'Bleep bloop')

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/site-packages/gitterpy/client.py in send(self, room_name, text)
    171         return self.post(
    172             self.set_message_url(room_id),
--> 173             data={'text': text}
    174         )
    175

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/site-packages/gitterpy/client.py in post(self, api, **kwargs)
     49
     50     def post(self, api, **kwargs):
---> 51         return self.request_process(r.post, api, **kwargs)
     52
     53     def put(self, api, **kwargs):

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/site-packages/gitterpy/client.py in request_process(self, method, api, **kwargs)
     43     def request_process(self, method, api, **kwargs):
     44         url = GITTER_BASE_URL + api
---> 45         return method(url, headers=self.headers, **kwargs).json()
     46
     47     def get(self, api, **kwargs):

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/models.py in json(self, **kwargs)
    890                     # used.
    891                     pass
--> 892         return complexjson.loads(self.text, **kwargs)
    893
    894     @property

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    352             parse_int is None and parse_float is None and
    353             parse_constant is None and object_pairs_hook is None and not kw):
--> 354         return _default_decoder.decode(s)
    355     if cls is None:
    356         cls = JSONDecoder

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py in decode(self, s, _w)
    337
    338         """
--> 339         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    340         end = _w(s, end).end()
    341         if end != len(s):

~/.pyenv/versions/3.6.4/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py in raw_decode(self, s, idx)
    355             obj, end = self.scan_once(s, idx)
    356         except StopIteration as err:
--> 357             raise JSONDecodeError("Expecting value", s, err.value) from None
    358         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The `JSONDecodeError` is not very indicative of what's wrong, so I suggest raising a `GitterRoomError` instead whenever a room is not found.

If you're concerned about breaking third-party use of `find_by_room_name()`, an alternative could be to set `raise_error_if_not_found` to `False` by default, and change all occurrences of `self.find_by_room_name(room_name)` to `self.find_by_room_name(room_name, True)` internally. Better perhaps, release a new major version where it's `True` by default?

### CODE REVIEW

**Comment by MichaelYusko** [comment]
@danmichaelo Hello, thank you for the report :+1: I'll review the PR today at night, and maybe we will need to make an small changes.

Thank you again!

**Comment by MichaelYusko** [comment]
@danmichaelo Reviewed, and everything looks good. But I have one comment
Could we change this line? https://github.com/MichaelYusko/GitterPy/pull/44/commits/c4f12f2a72d3e0e7341cf61ce83e7e023f191556#diff-438abf706174684d62df63d99dd61f14R82

**Comment by danmichaelo** [comment]
Yeah, I also thought about that, but I didn't manage to come up with something short that made much sense.. How about `raise_error`? *Or* perhaps have two methods?

``` python
    def find_by_room_name(self, name):
        room_id = ''
        for x in self.rooms_list:
            if x['name'] == name:
                room_id = x['id']
        return room_id

    def get_room(self, name):
        room_id = self.find_room_by_name(name)
        if room_id == '':
            raise GitterRoomError(name)
        return room_id
```

**Comment by MichaelYusko** [comment]
@danmichaelo Yeah, it's looks great, and when you will make an new changes, please make sure that  your branch is up-to date with the master branch. Have a nice day :+1:

**Comment by danmichaelo** [comment]
Sometimes it helps to think twice about something :)

File: gitterpy/client.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def find_by_room_name(self, name):
                 room_id = x['id']
         return room_id
 
+    def get_room(self, name):
+        room_id = self.get_room(name)
```
Comment (MichaelYusko) [inline gitterpy/client.py]: @danmichaelo In this line, maybe we need to use `find_by_room_name(name)`? instead `get_room(name)`

Comment (danmichaelo) [reply gitterpy/client.py]: That's a nice way to state it ;) Thx!

**Comment by MichaelYusko** [comment]
@danmichaelo Merged, nice work!

**Pull request merged**